core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/StopStartDownscaleDecommissionViaCMHandler.java

```diff
@@ -91,16 +97,36 @@ protected Selectable doAccept(HandlerEvent<StopStartDownscaleDecommissionViaCMRe
             HostGroup hostGroup = hostGroupService.getByClusterIdAndName(cluster.getId(), request.getHostGroupName())
                     .orElseThrow(NotFoundException.notFound("hostgroup", request.getHostGroupName()));
 
+            List<InstanceMetaData> instancesWithServicesNotRunning = cloudInstanceIdToInstanceMetaDataConverter.getNotDeletedAndNotZombieInstances(stack,
+                    request.getHostGroupName(),
+                    request.getRunningInstancesWithServicesNotRunning().stream()
+                            .map(CloudInstance::getInstanceId)
+                            .collect(Collectors.toSet()));
+
```

Comment:
Should we also be validating the state of the services on the nodes that were provided to us (i.e. not just the ones which were discovered to be in bad state). Probably part of a follow-up patch.